core/deployment/src/main/java/io/quarkus/deployment/dev/devservices/ComposeDevServicesBuildTimeConfig.java

```diff
@@ -0,0 +1,118 @@
+package io.quarkus.deployment.dev.devservices;
+
+import java.time.Duration;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import io.quarkus.runtime.annotations.ConfigGroup;
+import io.smallrye.config.WithDefault;
+
+@ConfigGroup
+public interface ComposeDevServicesBuildTimeConfig {
+
+    /**
+     * Compose dev service enabled or disabled
+     */
+    @WithDefault("true")
+    boolean enabled();
+
+    /**
+     * List of file paths relative to the project root for Compose dev service configuration,
+     * if not provided will look for compose files in the project root
+     */
+    Optional<List<String>> files();
+
+    /**
+     * Name of the compose project, used to discover running containers,
+     * if not provided a project name will be generated
+     */
+    Optional<String> projectName();
+
+    /**
+     * Compose profiles to activate
+     */
+    Optional<List<String>> profiles();
+
+    /**
+     * List of additional options to pass to compose command
+     */
+    Optional<List<String>> options();
+
+    /**
+     * Whether to run compose up and start containers at startup, when disabled, services are discovered by project name
+     */
+    @WithDefault("true")
+    boolean startServices();
+
+    /**
+     * Whether to run compose down and stop containers at shutdown
+     */
+    @WithDefault("true")
+    boolean stopServices();
+
+    /**
+     * Whether to use test containers Ryuk resource reaper to clean up containers
+     */
+    @WithDefault("true")
+    boolean ryukEnabled();
```

Comment:
Is this needed/necessary here or something to put as global thing for all container based devservices ?

Reply:
Test containers by default cleans up all containers created by itself using the resource reaper (ryuk) container. This is in addition to the shutdown hook extensions register to stop remove containers. 

There is an API, and you can give it a resource filter to clean up. It'll only create the ryuk container when used. 

Here containers are not created by testcontainers but scheduled by compose. So we can pass the compose project label as a resource filter for the resource reaper to cleanup. As before, this is in addition to the compose down we normally do.